src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -33,60 +30,87 @@ public Question(NsUser writer, String title, String contents) {
     public Question(Long id, NsUser writer, String title, String contents) {
         this.id = id;
         this.writer = writer;
-        this.title = title;
-        this.contents = contents;
+        this.detail = new Detail(title, contents);
     }
 
     public Long getId() {
         return id;
     }
 
-    public String getTitle() {
-        return title;
+    public NsUser getWriter() {
+        return writer;
     }
 
-    public Question setTitle(String title) {
-        this.title = title;
-        return this;
+    public void addAnswer(Answer answer) {
+        answers.add(answer, this);
     }
 
-    public String getContents() {
-        return contents;
+    public boolean isOwner(NsUser loginUser) {
+        return writer.equals(loginUser);
     }
 
-    public Question setContents(String contents) {
-        this.contents = contents;
-        return this;
+    public boolean isDeleted() {
+        return deleted;
     }
 
-    public NsUser getWriter() {
-        return writer;
-    }
+    public DeleteHistories delete(NsUser user) throws CannotDeleteException {
```

Comment:
```suggestion
    public void delete(NsUser user) throws CannotDeleteException {
```
바로 아래 DeleteHistories를 생성하는 별도의 메서드가 있다.
메서드가 한 가지 일만 한다는 원칙을 지켜 void로 반환하는 것은 어떨까?